### PR TITLE
Add static libs for x86/x86_64 Android

### PR DESCRIPTION
**Pull request by MarkY-LunarG**
There are some platforms that want to use the lz4 and zlib compression but may not have access to install their platform libraries. This is particularly true on Android, so add the supporting x86/x86_64 versions of static support libraries.

As with previous builds of static libraries, the versions of the source material are:
  - libz   : v1.2.11
    + Modified configure file "CFLAGS" to include "-fPIC"
    + Forced 32-bit build with modified "CFLAGS" to include "-m32"
  - liblz4 : v1.8.3
    + Built with CFLAGS predefined to "-fPIC"
    + Forced 32-bit build with modified "CFLAGS" to include "-m32"

### CODE REVIEW

**Comment by MarkY-LunarG** [comment]
Fixes issue #1078

**Review by andrew-lunarg — APPROVED** [review]
Looks good to me caveat whitespace issues, which I wouldn't block on if it seems fine to you, and also caveat I'm not the best with build system stuff and trust that those changes make sense and passing CI is enough of an endorsement.

As an observation for all of team, the folder organisation under Android and Linux seems more logical than separating at the top level for every supported windows variant, and maybe we could get to having one header copy per library rather than duplicating the headers for the different platform configs in some future PR.

File: framework/util/CMakeLists.txt [hunk 1]
```diff
@@ -170,13 +178,34 @@ if (TARGET ZLIB::ZLIB)
...
                                    $<$<BOOL:${WIN32}>:ZLIB_WINAPI>)
     target_link_libraries(gfxrecon_util ZLIB::ZLIB)
+elseif (USE_STATIC_LIBZ)
+    MESSAGE(INFO "Using precompiled static libz - ${STATIC_LIBZ}")
+    add_library(libz STATIC IMPORTED)
+    set_target_properties(libz PROPERTIES
+        IMPORTED_LOCATION "${STATIC_LIBZ}"
+    )
+    target_compile_definitions(gfxrecon_util
+                            PUBLIC
```
[the remark this answers is not in the document]
Comment (MarkY-LunarG) [reply framework/util/CMakeLists.txt]: Fixed.  And did a little more cleanup there as well.

**Comment by MarkY-LunarG** [comment]
Verified x86 Chromebook still builds with the latest changes.

**Comment by MarkY-LunarG** [comment]
Removed x86/x64 static Linux build changes which introduced issues with the Windows build.  Something we can add in the future if needed.

**Pull request merged**